packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -253,10 +253,10 @@ const CustomElementHook: Hooks<VCustomElement> = {
 };
 
 function linkNodeToShadow(elm: Node, owner: VM) {
-    const { shadowMode } = owner;
+    const { renderMode, shadowMode } = owner;
 
     // TODO [#1164]: this should eventually be done by the polyfill directly
-    if (shadowMode === ShadowMode.Synthetic) {
+    if (shadowMode === ShadowMode.Synthetic || renderMode === RenderMode.Light) {
```

Comment:
It is not obvious to me why we need to link the element to the shadow root when rendering a Light DOM component. Theoretically, this is not needed when synthetic shadow DOM is not loaded.